iob.js

```diff
@@ -0,0 +1,69 @@
+'use strict';
+function calcTotal(treatments, profile, time) {
+var iob = 0
+, activity = 0;
+if (!treatments) return {};
+if (profile === undefined) {
+//if there is no profile default to 3 hour dia
+profile = {dia: 3, sens: 0};
+}
+if (time === undefined) {
+time = new Date();
+}
+treatments.forEach(function (treatment) {
+if (new Date(treatment.created_at) < time) {
+var tIOB = calcTreatment(treatment, profile, time);
+if (tIOB && tIOB.iobContrib) iob += tIOB.iobContrib;
+if (tIOB && tIOB.activityContrib) activity += tIOB.activityContrib;
+}
+});
+return {
+iob: iob,
+display: iob.toFixed(2) == '-0.00' ? '0.00' : iob.toFixed(2),
+activity: activity
+};
+}
+function calcTreatment(treatment, profile, time) {
+var dia = profile.dia
+, scaleFactor = 3.0 / dia
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: ['scaleFactor' is defined but never used.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/762667217/issues/source?bid=2452374&fileBranchId=2452471#l28)